runtime/bcutil/cfreader.c

```diff
@@ -1635,6 +1635,21 @@ checkFields(J9PortLibrary* portLib, J9CfrClassFile * classfile, U_8 * segment, U
 			}
 		}
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_VALUE_TYPE)) {
+			if (J9_ARE_NO_BITS_SET(value, CFR_ACC_STATIC | CFR_ACC_FINAL)) {
```

Comment:
There is an existing check on `CFR_ACC_VALUE_TYPE` class that does not have `CFR_ACC_FINAL` set at:
https://github.com/eclipse-openj9/openj9/blob/194ae711901aeec136e9679c660c2b6a9be086cb/runtime/bcutil/cfreader.c#L2895-L2897

Reply:
Unless I misunderstand what you're saying, these checks seem to be completely different? The existing check is for the class itself, but the check I added is for the fields of the class. 

 It is for this line in the spec: _"In a value class, each field must have at least one of its ACC_STATIC or ACC_FINAL flags set."_ http://cr.openjdk.java.net/~dlsmith/jep401/jep401-20211220/specs/value-objects-jvms.html#jvms-4.5-200-A.1

Reply:
Ah, it is the field. I missed that.